pyproject.toml

```diff
@@ -99,7 +99,7 @@ timeout = 300
 # Restore this setting to debug failures
 # timeout_method = "thread"
 filterwarnings = [
-  "error",
+  "error:::jupyter.*",
```

Comment:
This is lazy, but catches packages that start with `jupyter`, which most of ours do. We could focus specifically on this package with `jupyter_server`, or explicitly list one or more exact packages we want to be strict with

Reply:
I think the right solution then is to update the contributing guide to show how to disable warnings (as we do with the minimum version test: `pytest -vv -W default`

Reply:
I think running the prerelease tests with -Werror is a good idea, since it serves a similar purpose: peeking into the future for things that _will_ fail, but haven't yet.

Reply:
Would you like me to take over this PR and update the workflows/docs?  I like the idea of preserving the errors for our own warnings, I think we should keep that change.